core/src/main/java/io/grpc/internal/DelayedClientTransport.java

```diff
@@ -80,44 +80,41 @@ public void start(Listener listener) {
   @Override
   public ClientStream newStream(MethodDescriptor<?, ?> method, Metadata headers) {
     Supplier<ClientTransport> supplier = transportSupplier;
+    if (supplier == null) {
+      synchronized (lock) {
+        // Check again, since it may have changed while waiting for lock
+        supplier = transportSupplier;
+        if (supplier == null && !shutdown) {
+          PendingStream pendingStream = new PendingStream(method, headers);
+          pendingStreams.add(pendingStream);
+          return pendingStream;
+        }
+      }
+    }
     if (supplier != null) {
```

Comment:
After `suppler != null`, we don't care about `shutdown`. In the docs for `shutdown()`:
> Prevents creating any new streams until {@link #setTransport} is called.

After `setTransport()` is called, `DelayedClientTransport` does full delegation ignoring `shutdown`. This can seem a bit strange, but it is a really useful feature for current code as it allows cleaning up old `DelayedClientTransports`, is easier to implement, and there aren't any usages yet that would benefit from stricter behavior.

We use the `dct.setTransport(realTransport); dct.shutdown();` pattern in both `ManagedChannelImpl` and `TransportSet`. Because of the race between `obtainActiveTransport()` returning and the caller creating a new stream, with a strict behavior it would be very hard to cleanup/shutdown a `DelayedClientTransport` without spurious failures.